### PR TITLE
[ocaml-base] put opambin into PATH before spawning merlin

### PR DESCRIPTION
**Pull request by yunxing**
Sometimes people don't run `opam config env` before starting Atom. Therefore opam's binary directory (something like `/Users/yunxing/.opam/4.02.3/bin`) may not be included in the search path and Nuclide cannot find merlin to run.

This patch forces opam to be part of PATH before running merlin. 

@jordwalke

### CODE REVIEW

File: pkg/nuclide/ocaml-base/lib/MerlinProcess.js [hunk 1]
```diff
@@ -284,12 +304,22 @@ function getMerlinFlags(): Array<string> {
 
 let isInstalledCache: ?boolean = null;
 async function isInstalled(merlinPath: string): Promise<boolean> {
+  opamBin = await getOpamBin();
```
Comment (jordwalke) [inline pkg/nuclide/ocaml-base/lib/MerlinProcess.js]: Should probably be a little utility function.

File: pkg/nuclide/ocaml-base/lib/MerlinProcess.js [hunk 2]
```diff
@@ -243,8 +257,14 @@ export async function getInstance(file: NuclideUri): Promise<?MerlinProcess> {
 
   const dotMerlinPath = await findNearestFile('.merlin', file);
 
+  opamBin = await getOpamBin();
+  env = process.env.PATH;
+  if (opamBin) {
+    env = env + ":" + opamBin
```
Comment (jordwalke) [inline pkg/nuclide/ocaml-base/lib/MerlinProcess.js]: Is there a cross platform way to join paths? I genuinely do not know.

Comment (ssorallen) [inline pkg/nuclide/ocaml-base/lib/MerlinProcess.js]: They can be joined like strings here, but the platform-specific delimiter should be used from Node's [`path.delimiter`](https://nodejs.org/api/path.html#path_path_delimiter):

```js
import path from 'path';

...

if (opamBin) {
  env = `${env}${path.delimiter}${opamBin}`;
}
```

**Comment by ssorallen** [comment]
Can you try linting your code? It looks like you're missing `const` or `let` on your variable declarations, which is something ESLint will catch.

```
$ cd /path/to/nuclide
$ npm install
$ npm run lint-all
```

File: pkg/nuclide/ocaml-base/lib/MerlinProcess.js (hunk 2 above)
Comment (yunxing) [inline pkg/nuclide/ocaml-base/lib/MerlinProcess.js]: Good point!

**Comment by yunxing** [comment]
@ssorallen @jordwalke Linted and fixed as suggested.

**Comment by yunxing** [comment]
@ssorallen @jordwalke Does this look good?

**Comment by ssorallen** [comment]
Looks good to me. Can you rebase onto latest master?

@jordwalke, look good to you?

**Comment by jordwalke** [comment]
Yeah awesome thanks.

**Comment by nmote** [comment]
@jordwalke @ssorallen are either of you planning on driving this through?

**Comment by nmote** [comment]
@yunxing if you rebase one more time I will drive this through. Sorry that we dropped the ball on this PR.

**Comment by nmote** [comment]
I'm still happy to make sure this gets in if you update. Feel free to reopen or create a new PR.

**Comment by jordwalke** [comment]
I think we've got an even better solution coming up! :D

**Pull request closed without merging**